### PR TITLE
Stripe Activation now does stuff

### PR DESCRIPTION
**Pull request by aramshiva**
Fixes #5090. Activating Cards now does anything.

### CODE REVIEW

File: app/controllers/stripe_cards_controller.rb
Comment (sampoder) [inline app/controllers/stripe_cards_controller.rb]:
```suggestion
    return redirect_back fallback_location: event_cards_new_path(event), flash: { error: "Organization is in Playground Mode" } if event.demo_mode?
```

or ~~Organisation~~

File: config/routes.rb
Comment (cjdenio) [inline config/routes.rb]: Heads up that `member` blocks change the format of path helpers (e.g. `stripe_card_freeze_path` becomes `freeze_stripe_card_path`)
Screenshot 2024-01-24 at 8 24 31 PM

Comment (aramshiva) [reply config/routes.rb]: Resolved!

**Comment by YodaLightsabr** [comment]
- [x] - Default cards to inactive
- [x] - Build out activation page to enter last4
- [x] - Remove activate button from card show page and add information about shipping
- [x] - Add activate method to take that response and activate the cards

**Comment by aramshiva** [comment]
Does anyone know where the card is being defaulted to activated when creating a card? I wasn't able to find it.

**Comment by sampoder** [comment]
Please test this but I think it’s here: https://github.com/hackclub/hcb/blob/6dd0d6b78c1578a956c99b54d3599eb071377e82/app/services/stripe_card_service/create.rb#L72

**Comment by aramshiva** [comment]
I wasnt able to find it for some reason, but it was right infront of me, I forgot to check card services! Thanks!

**Comment by aramshiva** [comment]
http://hcb.hackclub.com/stripe_cards/activation
^ change `hack.af/activate` on merge to this

File: app/mailers/stripe_card_mailer.rb
Comment (sampoder) [inline app/mailers/stripe_card_mailer.rb]:
```suggestion
         subject: "Your HCB card for #{@event.name} was lost in shipping."
```

Giving them the last four probably isn't that helpful if we're hiding it at all other times

File: config/routes.rb
Comment (sampoder) [inline config/routes.rb]: Could we make the path `/stripe_cards/activate` and then differentiate between GET / POST requests?

I bit like how we do with `/auth`:

```
get "auth", to: "users#auth"
post "auth", to: "users#auth_submit"
```

It's a small thing but I feel like that'd be cleaner?

File: app/controllers/stripe_cards_controller.rb
Comment (sampoder) [inline app/controllers/stripe_cards_controller.rb]: We can skip doing Pundit all together if we're just making the policy `true` :D 

```
skip_after_action :verify_authorized, only: [:activation]
```

This can go at the top it will do that for you

File: app/mailers/stripe_card_mailer.rb
Comment (aramshiva) [reply app/mailers/stripe_card_mailer.rb]: I talked to Ian about this and he thought it might be useful with HCB support iirc

File: app/controllers/stripe_cards_controller.rb
Comment (aramshiva) [reply app/controllers/stripe_cards_controller.rb]: oh, this is just a placeholder for now

File: app/controllers/stripe_cards_controller.rb
Comment (garyhtou) [inline app/controllers/stripe_cards_controller.rb]: since we're using `skip_authorization`, this line likely is not needed

File: app/views/stripe_cards/shipping.html.erb
Comment (garyhtou) [inline app/views/stripe_cards/shipping.html.erb]: thank you!!

File: config/routes.rb
Comment (garyhtou) [reply config/routes.rb]: i like this idea!

I'd also suggest to make the `POST` a **member** path.

`GET`: /stripe_cards/activate
`POST`: /stripe_cards/:last4/active (/stripe_cards/1234/activate)

File: app/controllers/stripe_cards_controller.rb
Comment (sampoder) [reply app/controllers/stripe_cards_controller.rb]: `activation` also doesn't seem to exist

File: app/controllers/stripe_cards_controller.rb
Comment (sampoder) [inline app/controllers/stripe_cards_controller.rb]: Can we keep this code comment? feels helpful

**Review by garyhtou — COMMENTED** [review]
this is good to merge after making the form look more like a card, and testing it!

**Comment by aramshiva** [comment]
### To-Dos

Before Merge
- [x] Upgrade UI to include realistic card
- [x] Handle current cards already in transit
- [ ] Change `hack.af`

After Merge
- [ ] Write a changelog post
- [ ] Ship in #ship and #hcb
- [x] Message @sarthaktexas for Twitter/X post.

File: app/controllers/stripe_cards/activation_controller.rb
Comment (garyhtou) [inline app/controllers/stripe_cards/activation_controller.rb]:
```suggestion
        # Handle cards issued before activation feature launch
        if @card.created_at < Date.new(2024, 2, 22)
```

**Review by garyhtou — APPROVED** [review]
yay!

**Pull request merged**
